.github/workflows/docs.yml

```diff
@@ -0,0 +1,43 @@
+name: Documentation
+
+on: [push, pull_request]
+
+jobs:
+  docs:
+    runs-on: ubuntu-20.04
+    container:
+      image: ubuntu:22.04
+    steps:
+      - name: Install dependencies
+        env:
+          DEBIAN_FRONTEND: noninteractive
+        run: |
+          apt-get update
+          apt-get upgrade -y
+          apt-get install -y --no-install-recommends \
+          make automake autoconf libtool gettext autopoint gcc \
+          gtk-doc-tools shared-mime-info desktop-file-utils gnome-desktop-testing xmlto fuse libflatpak-dev \
+          libglib2.0-dev libgeoclue-2-dev libjson-glib-dev libfontconfig1-dev libfuse-dev libportal-dev libpipewire-0.3-dev
+
+      - name: Check out xdg-desktop-portal
+        uses: actions/checkout@v2
+
+      - name: Build docs
```

Comment:
I changed the main build workflow to build docs itself. Does it make sense to just add the deploy step to one of those instead of having its own? Fewer duplicated places to maintain dependencies and whatnot.

Reply:
Or we could keep this and disable docs in the main workflow so for contributors it has a more clear result.

Reply:
I moved the docs part out of the checks jobs. Ideally, there would be some containers built that contain all the dependencies needed for the CI to run, but that's out of the scope of this PR & it seems to be fast enough, so it's fine I guess?

Reply:
Yeah, I'm ok with that.